### ISSUE TITLE
vllm has a new version

### ISSUE THREAD

**Issue by ChengYen-Tang** [blazux/qwen3.8-Flash-DGX#14]
https://github.com/vllm-project/vllm/releases/tag/v0.29.0

# Model Runner V2 (MRV2) Effects and Improvements in vLLM v0.29.0

## Core Enhancements

1. **Becomes the Default Runner** - MRV2 is now the **default runner for all models**, marking the completion of the rollout process that began in v0.28.0.

2. **CUDA Graph Memory Optimization**
   - New CUDA graph memory profiling capability
   - Used for KV cache auto-sizing

3. **Batch-Sharded Sampling** - Reduces per-step logits memory by **1/TP** (Tensor Parallelism factor)

4. **New Feature Support**
   - Prompt embeds
   - `extract_hidden_states` speculation
   - Padded FULL cudagraph dispatch for uniform decode under speculative decoding
   - DP-sync skipping (before EAGLE/MTP draft prefill)

5. **Encoder Path Optimization** - Encoder-only path split out separately

6. **Memory Management Improvements** - Correctly releases memory on shutdown and sleep

## Performance Impact

The rollout of MRV2 brings:
- **Lower memory consumption** (particularly logits memory)
- **Improved throughput** (through batch-sharded sampling)
- **Better KV cache management**

（Summary by Github copilot)

**Comment by blazux** [comment]
Thanks — checked. v0.29.0 ships Qwen3.8-Flash-Next natively (#53896) with an arm64 image, and already contains two of the fixes this repo patches (vllm#50729, vllm#52775). Still missing there: the prefix-caching block_size bug (core.py still takes the min group block size), the GB10 FLA shared-memory gate, the deterministic persistent_topk (vllm#55122 is still open), fp8 KV on the QSA path, and disable_eagle_block_drop; and the model package is now `vllm/models/qwen4_exp`, so every patch target moves. That makes it a port with a full re-validation, not an image bump.

**Comment by ChengYen-Tang** [comment]
Are there plans to upgrade vllm to version 0.29.0?

**Comment by blazux** [comment]
Hi, there is a plan to give it a try and run few benches on it. Since some fixes are still missing the upgrade will depends on the results (are the benefits worth the hassle?), but rest assured there is a plan to keep improving this recipe and it will obviously come with vLLM upgrade 😀

**Comment by blazux** [comment]
Update: the port is done and shipped as an option. Dockerfile.v0.29 builds the same recipe on vllm/vllm-openai:v0.29.0 (the model is qwen4_exp there): vllm#50729 and vllm#52775 are already in the release so those two patches are dropped; the prefix-caching block-size fix, the GB10 FLA gate, the deterministic top-k kernel, the hybrid dispatch and the reduced draft vocabulary were re-targeted; the PLE mmap patch was rewritten for the new layer. Not ported yet: the fp8 KV cache.


docker build -f Dockerfile.v0.29 -t qwen38-flash-dgx:v0.29 .
IMAGE=qwen38-flash-dgx:v0.29 MODE=hybrid YARN=1 CTX=500000 scripts/serve.sh
or ./flash serve v0.29. serve.sh reads the base from the image label, same knobs and defaults as the preview image.

Measured on our GX10 with the default recipe: decode 36.4 tok/s, prefill 2,529 / 3,026 tok/s (8k / 32k), needle at 92k in 45.4 s, deterministic at temperature 0, prefix-cache hits bit-exact, and on our agentic benchmark it scores exactly what the preview image scores (6 runs each side, same failures, within run-to-run variance). Draft acceptance is slightly lower (64.8% vs 68%), which we have not looked into yet.

It is now running on our own box for a long-term validation. If nothing shows up over the coming days and once fp8 KV is ported, Dockerfile.v0.29 becomes the default in the near future and the preview one moves to Dockerfile.preview. Details in the README section and docs/HOW-IT-WORKS.md. Thanks for the nudge!

**Comment by ChengYen-Tang** [comment]
Hi @blazux 
https://github.com/vllm-project/vllm/releases/tag/v0.30.0

Version 0.30.0 showed a 15% speed improvement on two RTX A6000 cards at my company. Want to give it a try?

**Comment by blazux** [comment]
@ChengYen-Tang 
I will give a try and see if everything is fine
